src/sentry/api/helpers/group_index/update.py

```diff
@@ -738,3 +724,32 @@ def update_groups(
         result["inbox"] = inbox
 
     return Response(result)
+
+
+def handle_has_seen(
+    has_seen: Any,
+    group_list: Sequence[Group],
+    group_ids: Sequence[Group],
+    project_lookup: Dict[int, Project],
+    projects: Sequence[Project],
+    acting_user: User | None,
+) -> None:
+    is_member_map = {
+        project.id: (
+            project.member_set.filter(user_id=acting_user.id).exists() if acting_user else False
+        )
+        for project in projects
+    }
+    user_id = acting_user.id if acting_user else None
+    if has_seen:
```

Comment:
```suggestion
    if has_seen is True:
```

Total stylistic nit, but I like the parallelity of either doing `if thing/if not thing` or `if thing is True/if thing is False`. (That said, this relies on `has_seen` being a boolean - see my question above.)

Reply:
> I'm trying to avoid changing any of the implementation with this refactor 

Fair point. And a lot of these functions you're splitting out have the same issues, so perhaps it's easier to just wait until they're all broken out and then just fix a given problem across all of them.